src/main/java/cart/authentication/AuthorizationExtractor.java

```diff
@@ -0,0 +1,8 @@
+package cart.authentication;
+
+import javax.servlet.http.HttpServletRequest;
+
+public interface AuthorizationExtractor<T> {
+
+    T extact(HttpServletRequest request);
```

Comment:
```suggestion
    T extract(HttpServletRequest request);
```
제네릭을 쓰신 이유가 있을까요? 🤔
basic 인증에서 다른 인증으로 변경되더라도 동일한 MemberInfo를 반환하도록 만들면 되지 않을까 싶어요.
계속해서 형변환을 할 필요는 없을 것 같습니다.

Reply:
저는 다양한 형태의 인증이 있을 수도 있다고 생각을 했었는데 지금 생각해보니 현재 이것은 인증을 위한 인터페이스이고 member를 검증하는 방식이니 memberInfo를 반환하는 것이 좋다고 생각합니다!